### PR TITLE
Adds a paper slip "id card" to the game

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a new paper slip item, which can be placed in your id card slot and written on to change your id's name, at the cost of no airlock access, bank account, and appearing very suspicious on secHUDS.

To make: 5 paper and a wirecutters in the crafting menu

## Why It's Good For The Game
Makes it easier to get rid of an unknown status without having to rely on the heads of staff issuing a new id card, fun ic play 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

paper

paper_hud

paper_examine

</details>

## Changelog
:cl:
add: Added a paper slip that acts as a fake id card that you can use to fake identity, craftable with paper and wirecutters.
/:cl:

### CODE REVIEW

**Comment by Miliviu** [comment]
I would add a Screenshot of the sprite to Testing Photographs and Procedure!

File: _maps/_basemap.dm
Comment (EvilDragonfiend) [inline _maps/_basemap.dm]: I think you forgot this

File: beestation.dme
Comment (EvilDragonfiend) [inline beestation.dme]: and this too

File: code/game/objects/items/cards_ids.dm
Comment (Miliviu) [inline code/game/objects/items/cards_ids.dm]: Vaugue?

File: _maps/_basemap.dm
Comment (etherware-novice) [reply _maps/_basemap.dm]: oh! i mustve pushed that accidentily

File: code/game/objects/items/cards_ids.dm
Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: wdym by vauge? sorry

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm]: I think he meant `vauge` that looks like a typo of `vague`

Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: oh i didnt notice that

File: code/game/objects/items/cards_ids.dm
Comment (EvilDragonfiend) [inline code/game/objects/items/cards_ids.dm]: this adds a lot of meaningless var to all types of cards, while this function is too simple.
I think you can just override new `Initialize()` than adding a new var.

Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: how would i do that?

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm]: ah, my bad. I thought it was from `Initialize()` not from `examine()`

Okay, don't use `if(!noaccount)`. use `if(!istype(src, /obj/item/card/id/paper))` instead.

**Comment by EvilDragonfiend** [comment]
also, you need more test procedure evidence. post images showing how sec hud is, and how examine text is.

**Comment by etherware-novice** [comment]
> also, you need more test procedure evidence. post images showing how sec hud is, and how examine text is.

how do i add procedure evidence?

**Comment by EvilDragonfiend** [comment]
> > also, you need more test procedure evidence. post images showing how sec hud is, and how examine text is.
> 
> how do i add procedure evidence?

just post image like you posted paper image.
It's unclear that it really shows proper examine lines and sec hud.

**Comment by EvilDragonfiend** [comment]
and make PDA, PDA painter, HoP console rejects them.

File: code/game/objects/items/cards_ids.dm
Comment (EvilDragonfiend) [inline code/game/objects/items/cards_ids.dm]: (nevermind the comment I left before. I think it's fine.)

File: code/game/machinery/PDApainter.dm
Comment (Zonespace27) [inline code/game/machinery/PDApainter.dm]: `elif` isn't what you mean to use

File: code/game/machinery/PDApainter.dm
Comment (Zonespace27) [inline code/game/machinery/PDApainter.dm]:
```suggestion
			to_chat(user, "<span class='warning'>There's no circuitry to change!")
```

File: code/game/machinery/computer/card.dm
Comment (Zonespace27) [inline code/game/machinery/computer/card.dm]:
```suggestion
			to_chat(user, to_chat(user, "<span class='warning'>There's no circuitry to change!"))
```

File: code/game/objects/items/cards_ids.dm
Comment (Zonespace27) [inline code/game/objects/items/cards_ids.dm]: early retunr

File: code/game/objects/items/cards_ids.dm
Comment (Zonespace27) [inline code/game/objects/items/cards_ids.dm]:
```suggestion
	to_chat(user, "You can't insert money into a slip!")  // not sure if this is triggerable but just as a safeclip
```

File: code/game/objects/items/cards_ids.dm
Comment (Zonespace27) [inline code/game/objects/items/cards_ids.dm]:
```suggestion
	to_chat(user, "There's no money circuitry in here!")
```

File: code/game/machinery/PDApainter.dm
Comment (etherware-novice) [reply code/game/machinery/PDApainter.dm]: oops

File: code/game/objects/items/cards_ids.dm
Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: its supposed to hide the "there is no registered account" str because you CANT link a account and it might be confusing

**Comment by EvilDragonfiend** [comment]
seems alright to me.

**Comment by itsmeow** [comment]
What does it look like on a sechud? Should be in testing screenshots

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [reply code/game/objects/items/cards_ids.dm]: what they're saying is instead of wrapping all this in an if, do
`if(istype(src, ...))
return
`
that way you make less code impact as per our contributing guidelines

**Comment by EvilDragonfiend** [comment]
uh, It looks it is still acceptable to PDA?

**Comment by EvilDragonfiend** [comment]
and i am not sure if you intended this - you can change your voice without the help of syndicate agent card forging when you have a voice changing mask.

**Comment by etherware-novice** [comment]
> and i am not sure if you intended this - you can change your voice without the help of syndicate agent card forging when you have a voice changing mask.

i figured it could still do that because consider you still have no access at all and anyone who inspects you will instantly notice you dont have a card which might give you away

File: code/game/objects/items/cards_ids.dm
Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: oh! my mistake

**Comment by EvilDragonfiend** [comment]
> > and i am not sure if you intended this - you can change your voice without the help of syndicate agent card forging when you have a voice changing mask.
> 
> i figured it could still do that because consider you still have no access at all and anyone who inspects you will instantly notice you dont have a card which might give you away

No, you can be a tcomms agent with less paying 2 tc than before.

**Comment by etherware-novice** [comment]
fixed

**Comment by LodedDiper69** [comment]
I don't think you have properly explained why this is good for the game. Sure, it makes faking an ID slightly easier, but don't we have agent cards already? What does this exactly contribute towards gameplay?

**Comment by etherware-novice** [comment]
@LodedDiper69 sorry! its basically supposed to be a temporary card if yours was stolen lost or exploded. basically something to slap on so people dont question why an Unknown is walking around

**Comment by etherware-novice** [comment]
seems the computer code changed heh

File: code/game/objects/items/cards_ids.dm
Comment (Crossedfall) [inline code/game/objects/items/cards_ids.dm]:
```suggestion
		name = "[(!newname)	? "paper slip identifier": "[newname]'s paper slip"]"
```

We can see that they're wearing the slip, having presumably here doesn't make sense.

File: code/game/objects/items/cards_ids.dm
Comment (Crossedfall) [inline code/game/objects/items/cards_ids.dm]:
```suggestion
	name = "[(!registered_name)	? "paper slip identifier": "[registered_name]'s paper slip"]"
```

^

**Comment by itsmeow** [comment]
> seems the computer code changed heh

you're welcome 😈

File: code/game/machinery/computer/card.dm
Comment (Crossedfall) [inline code/game/machinery/computer/card.dm]: Looks like you have an extra bit of whitespace here causing checks to fail.

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: Should be a variable on the ID basetype saying if bank accounts are restricted rather than checking type (See all other instances of specifically checking for type).

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm]: I am going to refactor this too once the paper slip is merged (from chameleon ID PR one). I think you can try to merge and then see what I would do - if you don't have anything else to ask for a change at this PR.

Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: so should i adjust it or wait until its merged and dragon fixes it

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm]: let's see what Bacon would say

Comment (PowerfulBacon) [reply code/game/objects/items/cards_ids.dm]: I think it should just be a var on the ID

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: I think this could just be something that indicates that its an electronic ID and all cases of type checking can be removed.

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: Lacks \<span>s

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: Lacks \<span>s

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: Lacks \<span>s

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: I think this message is better off being removed

File: code/game/objects/items/cards_ids.dm
Comment (PowerfulBacon) [inline code/game/objects/items/cards_ids.dm]: Should be called something like paper name tag rather than identifier.

File: code/modules/mob/living/carbon/human/say.dm
Comment (PowerfulBacon) [inline code/modules/mob/living/carbon/human/say.dm]: I think the ID paper should work for this, considering its the mask doing the voice changing and has nothing to do with the ID

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/mob/living/carbon/human/say.dm
Comment (etherware-novice) [reply code/modules/mob/living/carbon/human/say.dm]: someone else told me to explicitly remove it for balance reasons

Comment (EvilDragonfiend) [reply code/modules/mob/living/carbon/human/say.dm]: >  for balance reasons

No, I was asking if you know what you're doing

Comment (etherware-novice) [reply code/modules/mob/living/carbon/human/say.dm]: i think i know what im doing?? should i keep the id not working or

Comment (EvilDragonfiend) [reply code/modules/mob/living/carbon/human/say.dm]: Nah, literally was asking. You're taking my words more than you need to understand enough. It is fine as long as there's no not-intentional glitch.

Comment (etherware-novice) [reply code/modules/mob/living/carbon/human/say.dm]: im confused- should i keep the id card not working with the voicechanger??

Comment (EvilDragonfiend) [reply code/modules/mob/living/carbon/human/say.dm]: It is up to you at this point, and I was only letting you know the paper id is compatible with the syndi voice changer. What Bacon said is suggesting, not more than you must reflect his opinion. If you think it shouldn't work with the voice changer through the paper ID, then say your opinion here. That's it.

File: code/game/machinery/PDApainter.dm
Comment (itsmeow) [inline code/game/machinery/PDApainter.dm:171]: PDA painters don't affect circuitry. Just make it return and say nothing.

File: code/game/machinery/computer/card.dm
Comment (itsmeow) [inline code/game/machinery/computer/card.dm]: Missing `</span>`

File: code/game/machinery/PDApainter.dm
Comment (itsmeow) [inline code/game/machinery/PDApainter.dm]: PDA painters don't affect circuitry. Just make it return and say nothing.

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm:729]: Use the DEFINES here `MELEE`, etc

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm]: Shouldn't this be `Unknown`?

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm]: Also `Unknown`

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm]: How about something else (the pass has a speaker? and can talk)

How about: "The paper swipes along the pass. Nothing happens."

Also, missing `</span>`

File: code/game/objects/items/devices/PDA/PDA.dm
Comment (itsmeow) [inline code/game/objects/items/devices/PDA/PDA.dm]: "You attempt to jam the paper into your PDA's ID slot. It doesn't fit."

File: code/game/machinery/PDApainter.dm
Comment (itsmeow) [reply code/game/machinery/PDApainter.dm]: Also no `</span>`

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/cards_ids.dm
Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm:729]: literally every armor define uses this exact syntax i just copy pasted?

File: code/game/objects/items/devices/PDA/PDA.dm
Comment (itsmeow) [inline code/game/objects/items/devices/PDA/PDA.dm]: this doesn't return

File: code/game/machinery/PDApainter.dm
Comment (itsmeow) [inline code/game/machinery/PDApainter.dm]: electric

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm]: electric

File: code/modules/mob/living/carbon/human/say.dm
Comment (itsmeow) [inline code/modules/mob/living/carbon/human/say.dm]: this is intended? you can change your voice only with the paper slip?

File: code/game/machinery/computer/card.dm
Comment (itsmeow) [inline code/game/machinery/computer/card.dm]: you shouldn't need this, id_insert calls try_insert, right?

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [reply code/game/objects/items/cards_ids.dm:729]: there are defines, a lot of armor is out of date

Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm:729]: so....where is this and how do i define it

Comment (itsmeow) [reply code/game/objects/items/cards_ids.dm:729]: the defines already exist, literally just change the strings to `MELEE` etc

File: code/modules/mob/living/carbon/human/say.dm
Comment (etherware-novice) [reply code/modules/mob/living/carbon/human/say.dm]: ah wait was adding the not op to everything oops

File: code/game/machinery/computer/card.dm
Comment (etherware-novice) [reply code/game/machinery/computer/card.dm]: well it doesnt hurt to have a fallback

Comment (itsmeow) [reply code/game/machinery/computer/card.dm]: the id_insert text is objectively better, "There's no circuitry to change" is a lot more vague. So it should use that if it is supposed to.

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/game/machinery/computer/card.dm
Comment (itsmeow) [inline code/game/machinery/computer/card.dm]: this is not addressed because it still returns and thus won't call id_insert

Please test with modular computer card insert and see if it actually says the other message.

Otherwise, remove the return and see if it says it then.

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by itsmeow** [comment]
New changes look good, can you confirm the behavior in testing if you try to insert it into a tablet?

How about the HoP console?

**Comment by etherware-novice** [comment]
yep those work fine (as in blocking it)

**Comment by itsmeow** [comment]
CL should be more descriptive

File: code/game/objects/items/cards_ids.dm
Comment (EvilDragonfiend) [inline code/game/objects/items/cards_ids.dm]: you override the original proc incorrectly.

**Comment by EvilDragonfiend** [comment]
and like itsmeow said, CL should be more descriptive.

File: code/game/objects/items/cards_ids.dm
Comment (etherware-novice) [reply code/game/objects/items/cards_ids.dm]: whats the proper way to do it

Comment (EvilDragonfiend) [reply code/game/objects/items/cards_ids.dm]: why are you even trying to override it? I see this code has no reason to exist.

**Comment by PowerfulBacon** [comment]
Unresolved review from EvilDragon

File: code/game/objects/items/cards_ids.dm
Comment (itsmeow) [reply code/game/objects/items/cards_ids.dm]: /obj/item/card/id/**paper**/set_hud_icon_on_spawn

**Comment by itsmeow** [comment]
Will approve if you can show the paper ID working on sechuds and update the CL to be more descriptive/sensible.

**Comment by etherware-novice** [comment]
> Will approve if you can show the paper ID working on sechuds (with all the current changes) and update the CL to be more descriptive/sensible.

second image

**Review by itsmeow — APPROVED** [review]

**Pull request merged**
